### DISCUSSION TITLE
Why there isn't Karma JS test runner integration on allurereport.org

### DISCUSSION THREAD

**Discussion by mourav2222** [allure-framework/allure2#2516] [Questions & Support] [answered]
Hello,

We use the karma-runner JS framework for testing angular framework. 
I want to use allure report for a test reporting, but i didn't find the karma JS integration on https://allurereport.org/modules/?category=JavaScript

There is karma-allure-reporter implementation https://www.npmjs.com/package/karma-allure-reporter, 
but it is old, last update 4 years ago and is missing on official allure report site. 

Why is it karma not supported, or other tips, what can i do in this situation?
On this site i don't find any discussions with questions about karma (search: karma) 

only here
https://github.com/allure-framework/allure-js/issues/404
Open: Add allure-karma-reporter to this monorepo #404

Other links about karma allure reporter in web
https://www.npmjs.com/package/karma-allure-reporter
https://github.com/allure-framework/allure-karma
https://github.com/allure-examples/allure-karma-example

**Comment by baev** [comment, marked as answer]
According to the readme in the official repo, [Karma is deprecated and is not accepting new features or general bug fixes.](https://github.com/karma-runner/karma?tab=readme-ov-file#karma-is-deprecated-and-is-not-accepting-new-features-or-general-bug-fixes)

[Allure Karma](https://github.com/allure-framework/allure-karma) integration has [relatively small amounts of usage](https://www.npmjs.com/package/karma-allure-reporter) and has almost zero interest from our community during last four years. 

Considering that, we also decided to deprecate official Allure Karma integration. For the users that still use Karma, you can keep using old integration versions or implement their integration with Allure themselfs.

**Comment by mourav2222** [comment]
@baev 
Thanks for explanation. i could integrate allure report with your package karma-allure-reporter
https://www.npmjs.com/package/karma-allure-reporter
